examples/combined_ODNP.py

```diff
@@ -57,54 +50,100 @@
 myinput = input("Look ok?")
 if myinput.lower().startswith("n"):
     raise ValueError("you said no!!!")
-# }}}
-# {{{ these change if we change the way the data is saved
-IR_postproc = "spincore_IR_v1"
-Ep_postproc = "spincore_ODNP_v3"
-# }}}
 powers = 1e-3 * 10 ** (dB_settings / 10.0)
-fl = psp.figlist_var()
-save_file = True
-nPoints = int(acq_time_ms * SW_kHz + 0.5)
-acq_time_ms = nPoints / SW_kHz
-pad_us = 0
-Ep_ph1_cyc = psp.r_[0, 1, 2, 3]
-IR_ph1_cyc = psp.r_[0, 2]
-IR_ph2_cyc = psp.r_[0, 2]
-date = datetime.now().strftime("%y%m%d")
+# }}}
+#{{{phase cycling
+Ep_ph1_cyc = r_[0, 1, 2, 3]
+total_points = len(Ep_ph1_cyc)*nPoints
+assert total_points < 2**14, "For Ep: You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+IR_ph1_cyc = r_[0, 2]
+IR_ph2_cyc = r_[0, 2]
+total_pts = len(IR_ph2_cyc)*len(IR_ph1_cyc)*nPoints
+assert total_pts < 2**14, "For IR: You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+#}}}
 # {{{ check for file
-myfilename = date + "_" + output_name + ".h5"
-if os.path.exists(myfilename):
+filename_out = filename + ".h5"
+if os.path.exists(filename_out):
     raise ValueError(
-        "the file %s already exists, so I'm not going to let you proceed!" % myfilename
+        "the file %s already exists, so I'm not going to let you proceed!" % filename_out
     )
 # }}}
+# {{{Collect Thermals
+control_thermal = run_spin_echo(
+    nScans=parser_dict['thermal_nScans'],
+    indirect_idx=0,
+    indirect_len=len(powers) + 1,
+    ph1_cyc=Ep_ph1_cyc,
+    adcOffset=parser_dict['adc_offset'],
+    carrierFreq_MHz=parser_dict['carrierFreq_MHz'],
+    nPoints=nPoints,
+    nEchoes=parser_dict['nEchoes'],
+    p90_us=parser_dict['p90_us'],
+    repetition=parser_dict['repetition_us'],
+    tau_us=parser_dict['tau_us'],
+    SW_kHz=parser_dict['SW_kHz'],
+    output_name=filename,
+    indirect_fields=("start_times", "stop_times"),
+    ret_data=None,
+)  # assume that the power axis is 1 longer than the
+#                         "powers" array, so that we can also store the
+#                         thermally polarized signal in this array (note
+#                         that powers and other parameters are defined
+#                         globally w/in the script, as this function is not
+#                         designed to be moved outside the module
+SpinCore_pp.stopBoard();
```

Comment:
collect a thermal with B12 powered off (still has USB plugged in but the thermal for the E(p) will turn it back on once done running)